Library/Homebrew/formula_installer.rb

```diff
@@ -144,7 +146,7 @@ def pour_bottle?(install_bottle_options = { warn: false })
 
   def install_bottle_for?(dep, build)
     return pour_bottle? if dep == formula
-    return false if Homebrew.args.build_formula_from_source?(dep)
+    return false if args&.build_formula_from_source?(dep)
```

Comment:
I don't like that there's a bunch of `nil` `args` around now. I think a `NullArgs` object or something would make the API nicer.